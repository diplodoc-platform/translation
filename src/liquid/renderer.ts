import {Token, TokenType} from 'src/liquid';

import {generateX} from 'src/xlf/generator';

export type RendererRuleSet = Record<TokenType | string, RendererRule>;

export type RendererRule = (token: Token) => string;

class Renderer {
    private rendered: string;
    private tokens: Token[];
    private rules: RendererRuleSet;

    constructor(tokens: Token[]) {
        this.tokens = tokens;
        this.rendered = '';

        this.rules = {
            Text: this.plainText.bind(this),
            Space: this.plainText.bind(this),
            If: this.wrapInX.bind(this),
            Else: this.wrapInX.bind(this),
            EndIf: this.wrapInX.bind(this),
            ForInLoop: this.wrapInX.bind(this),
            EndForInLoop: this.wrapInX.bind(this),
            Function: this.wrapInX.bind(this),
            Filter: this.wrapInX.bind(this),
        };
    }

    render() {
        for (const token of this.tokens) {
            const rule = this.rules[token.type];
            if (!rule) {
                throw new Error(`unexpected token: ${token.type}`);
            }

            this.rendered += rule(token);
        }

        return this.rendered;
    }

    private plainText(token: Token) {
        return token.value;
    }

    private wrapInX(token: Token) {
        const {type, value} = token;

        return generateX({
            ctype: `liquid_${type}`,
            equivText: value,
        });
    }
}

export {Renderer};
export default {Renderer};
