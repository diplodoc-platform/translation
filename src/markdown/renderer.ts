import MarkdownIt from 'markdown-it';
import {
    mdRenderer,
    MarkdownRendererParams,
    MarkdownRendererEnv,
    MarkdownRenderer,
} from '@diplodoc/markdown-it-markdown-renderer';

// configure with diplodoc plugins
// @ts-ignore
import meta from 'markdown-it-meta';
// @ts-ignore
import sup from 'markdown-it-sup';
import notes from '@doc-tools/transform/lib/plugins/notes';
import cut from '@doc-tools/transform/lib/plugins/cut';
import checkbox from '@doc-tools/transform/lib/plugins/checkbox';
import anchors from '@doc-tools/transform/lib/plugins/anchors';
import monospace from '@doc-tools/transform/lib/plugins/monospace';
import imsize from '@doc-tools/transform/lib/plugins/imsize';
import file from '@doc-tools/transform/lib/plugins/file';
import includes from '@doc-tools/transform/lib/plugins/includes';
import tabs from '@doc-tools/transform/lib/plugins/tabs';
import video from '@doc-tools/transform/lib/plugins/video';
import table from '@doc-tools/transform/lib/plugins/table';

import markdownHandlers, {MarkdownHandlersState} from './handlers';
import hooks, {HooksParameters} from './hooks';
import {rules} from './rules';

export type MarkdownRendererState = MarkdownHandlersState;

export type RenderParameters = {
    skeleton: string;
    translations: Map<string, string>;
} & DiplodocParameters;

export type DiplodocParameters = {
    lang?: string;
};

function render(parameters: RenderParameters) {
    if (!validParameters(parameters)) {
        throw new Error('invalid parameters');
    }

    const md = new MarkdownIt({html: true}) as HooksParameters['markdownit'];

    const {handlers, initState} = markdownHandlers.generate(parameters);
    const markdownHooks = hooks.generate({markdownit: md});

    // todo: add non-destructive way of extending markdown-renderer hooks
    for (const defaultHook of Object.entries(MarkdownRenderer.defaultHooks)) {
        const [lifecycle, hooks_] = defaultHook as any;

        if (markdownHooks.hooks[lifecycle]) {
            markdownHooks.hooks[lifecycle] = [...markdownHooks.hooks[lifecycle], ...hooks_];
        }
    }

    const mdOptions: MarkdownRendererParams<MarkdownHandlersState> = {
        handlers,
        initState,
        hooks: markdownHooks.hooks,
        rules,
    };
    const diplodocOptions = {
        lang: parameters.lang ?? 'ru',
        path: '',
    };

    const env: MarkdownRendererEnv = {
        source: parameters.skeleton.split('\n'),
    };

    // diplodoc plugins
    md.use(meta);
    md.use(notes, diplodocOptions);
    md.use(cut, diplodocOptions);
    md.use(sup, diplodocOptions);
    md.use(checkbox, diplodocOptions);
    md.use(anchors, diplodocOptions);
    md.use(monospace, diplodocOptions);
    md.use(imsize, diplodocOptions);
    md.use(file, diplodocOptions);
    md.use(includes, diplodocOptions);
    md.use(tabs, diplodocOptions);
    md.use(video, diplodocOptions);
    md.use(table, diplodocOptions);

    md.use(mdRenderer, mdOptions);

    return md.render(parameters.skeleton, env);
}

function validParameters(parameters: RenderParameters) {
    const {skeleton, translations} = parameters;

    const skeletonCondition = skeleton !== undefined;
    const translationsCondition = translations !== undefined;

    const conditions = [
        skeletonCondition,
        translationsCondition,
        validDiplodocParameters(parameters),
    ];

    return conditions.reduce((a, v) => a && v, true);
}

function validDiplodocParameters(parameters: DiplodocParameters) {
    const {lang} = parameters;

    return lang === undefined || lang === 'ru' || lang === 'en';
}

export {render, validParameters, validDiplodocParameters};
export default {render, validParameters, validDiplodocParameters};
