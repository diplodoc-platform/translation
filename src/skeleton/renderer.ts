import MarkdownIt from 'markdown-it';
import {
    mdRenderer,
    MarkdownRenderer,
    MarkdownRendererParams,
    MarkdownRendererEnv,
} from '@diplodoc/markdown-it-markdown-renderer';

// configure with diplodoc plugins
// @ts-ignore
import meta from 'markdown-it-meta';
// @ts-ignore
import sup from 'markdown-it-sup';
import notes from '@doc-tools/transform/lib/plugins/notes';
import cut from '@doc-tools/transform/lib/plugins/cut';
import checkbox from '@doc-tools/transform/lib/plugins/checkbox';
import anchors from '@doc-tools/transform/lib/plugins/anchors';
import monospace from '@doc-tools/transform/lib/plugins/monospace';
import imsize from '@doc-tools/transform/lib/plugins/imsize';
import file from '@doc-tools/transform/lib/plugins/file';
import includes from '@doc-tools/transform/lib/plugins/includes';
import tabs from '@doc-tools/transform/lib/plugins/tabs';
import video from '@doc-tools/transform/lib/plugins/video';
import table from '@doc-tools/transform/lib/plugins/table';

import skeletonHandlers from './handlers';
import hooks, {HooksParameters, HooksState} from './hooks';
import {rules} from './rules';

export type SkeletonRendererState = HooksState;

export type RenderParameters = BaseParameters & DiplodocParameters;
export type BaseParameters = {
    markdown: string;
};
export type DiplodocParameters = {
    lang?: string;
};

function render(parameters: RenderParameters) {
    if (!validParameters(parameters)) {
        throw new Error('invalid parameters');
    }

    const {markdown, lang} = parameters;

    const md = new MarkdownIt({html: true}) as HooksParameters['markdownit'];
    const env: MarkdownRendererEnv = {source: markdown.split('\n')};

    const {handlers} = skeletonHandlers.generate();

    const skeletonHooks = hooks.generate({markdownit: md});

    // todo: add non-destructive way of extending markdown-renderer hooks
    for (const defaultHook of Object.entries(MarkdownRenderer.defaultHooks)) {
        const [lifecycle, hooks_] = defaultHook as any;

        if (skeletonHooks.hooks[lifecycle]) {
            skeletonHooks.hooks[lifecycle] = [...skeletonHooks.hooks[lifecycle], ...hooks_];
        }
    }

    const mdOptions: MarkdownRendererParams<HooksState> = {
        handlers,
        initState: skeletonHooks.initState,
        rules,
        hooks: skeletonHooks.hooks,
    };
    const diplodocOptions = {
        lang: lang ?? 'ru',
        path: '',
    };

    // diplodoc plugins
    md.use(meta, diplodocOptions);
    md.use(notes, diplodocOptions);
    md.use(cut, diplodocOptions);
    md.use(sup, diplodocOptions);
    md.use(checkbox, diplodocOptions);
    md.use(anchors, diplodocOptions);
    md.use(monospace, diplodocOptions);
    md.use(imsize, diplodocOptions);
    md.use(file, diplodocOptions);
    md.use(includes, diplodocOptions);
    md.use(tabs, diplodocOptions);
    md.use(video, diplodocOptions);
    md.use(table, diplodocOptions);

    md.use(mdRenderer, mdOptions);

    return md.render(markdown, env);
}

function validParameters(parameters: RenderParameters) {
    const {markdown, lang} = parameters;

    const markdownCondition = markdown !== undefined;
    const langCondition = lang === undefined || lang === 'ru' || lang === 'en';
    const conditions = [markdownCondition, langCondition];

    return conditions.reduce((a, v) => a && v, true);
}

export {render, validParameters};
export default {render, validParameters};
