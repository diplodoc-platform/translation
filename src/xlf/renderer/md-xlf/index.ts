import MarkdownIt from 'markdown-it';
import {MarkdownRenderer, MarkdownRendererEnv} from '@diplodoc/markdown-it-markdown-renderer';
import {
    CustomRendererHooks,
    CustomRendererParams,
    customRenderer,
} from '@diplodoc/markdown-it-custom-renderer';

// configure with diplodoc plugins
// @ts-ignore
import meta from 'markdown-it-meta';
// @ts-ignore
import sup from 'markdown-it-sup';
import notes from '@diplodoc/transform/lib/plugins/notes';
import cut from '@diplodoc/transform/lib/plugins/cut';
import checkbox from '@diplodoc/transform/lib/plugins/checkbox';
import anchors from '@diplodoc/transform/lib/plugins/anchors';
import monospace from '@diplodoc/transform/lib/plugins/monospace';
import imsize from '@diplodoc/transform/lib/plugins/imsize';
import file from '@diplodoc/transform/lib/plugins/file';
import includes from '@diplodoc/transform/lib/plugins/includes';
import tabs from '@diplodoc/transform/lib/plugins/tabs';
import video from '@diplodoc/transform/lib/plugins/video';
import table from '@diplodoc/transform/lib/plugins/table';

import {XLFRendererState, xlfInitState} from './state';
import hooks, {HooksParameters} from './hooks';
import {handlers} from './handlers';
import rules from './rules';

import {TemplateParameters, generateTemplate, templateValidParameters} from 'src/xlf/generator';

export type RenderParameters = TemplateParameters & DiplodocParameters & BaseParameters;
export type BaseParameters = {
    markdown: string;
};
export type DiplodocParameters = {
    lang?: string;
};

function render(parameters: RenderParameters) {
    if (!validParameters(parameters)) {
        throw new Error('invalid parameters');
    }

    const wrapper = generateTemplate(parameters);

    const xlfRenderer = new MarkdownIt({html: true}) as HooksParameters['markdownit'];
    const xlfRules = rules.generate();
    const xlfHooks: {hooks: CustomRendererHooks} = hooks.generate({
        template: wrapper.template,
        markdownit: xlfRenderer,
    });

    // todo: add non-destructive way of extending custom-renderer hooks
    for (const lifecycleHook of Object.entries(MarkdownRenderer.defaultHooks)) {
        // eslint-disable-next-line @typescript-eslint/no-explicit-any
        const [lifecycle, hooks_] = lifecycleHook as any;

        const hooksForLifeCycle = xlfHooks.hooks[lifecycle];

        if (Array.isArray(hooksForLifeCycle)) {
            xlfHooks.hooks[lifecycle] = [...hooksForLifeCycle, ...hooks_];
        } else {
            xlfHooks.hooks[lifecycle] = [hooksForLifeCycle, ...hooks_];
        }
    }

    const initState = () => ({
        ...xlfInitState(wrapper),
        ...xlfRules.initState(),
    });

    const xlfOptions: CustomRendererParams<XLFRendererState> = {
        rules: xlfRules.rules,
        hooks: xlfHooks.hooks,
        initState,
        handlers,
    };

    const diplodocOptions = {
        lang: parameters.lang ?? 'ru',
        path: '',
    };

    const env: MarkdownRendererEnv = {
        source: parameters.markdown.split('\n'),
    };

    // diplodoc plugins
    xlfRenderer.use(meta);
    xlfRenderer.use(notes, diplodocOptions);
    xlfRenderer.use(cut, diplodocOptions);
    xlfRenderer.use(sup, diplodocOptions);
    xlfRenderer.use(checkbox, diplodocOptions);
    xlfRenderer.use(anchors, diplodocOptions);
    xlfRenderer.use(monospace, diplodocOptions);
    xlfRenderer.use(imsize, diplodocOptions);
    xlfRenderer.use(file, diplodocOptions);
    xlfRenderer.use(includes, diplodocOptions);
    xlfRenderer.use(tabs, diplodocOptions);
    xlfRenderer.use(video, diplodocOptions);
    xlfRenderer.use(table, diplodocOptions);

    xlfRenderer.use(customRenderer, xlfOptions);

    return xlfRenderer.render(parameters.markdown, env);
}

function validParameters(parameters: RenderParameters) {
    const {lang, markdown} = parameters;

    const markdownCondition = markdown !== undefined;
    const langCondition = lang === undefined || lang === 'ru' || lang === 'en';
    const conditions = [templateValidParameters(parameters), markdownCondition, langCondition];

    return conditions.reduce((a, v) => a && v, true);
}

export {render, validParameters};
export default {render, validParameters};
