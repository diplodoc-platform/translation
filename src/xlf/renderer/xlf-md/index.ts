import assert from 'assert';
import {XLFTagToken, XLFTextToken, XLFToken, isXLFTagToken, isXLFTextToken} from 'src/xlf/token';

export type XLFMDRendererRuleSet = {
    text: XLFMDRendererRule;
    [key: string]: XLFMDRendererRule;
};

export type XLFMDRendererRule = (token: XLFToken) => string;

class XLFMDRenderer {
    rules: XLFMDRendererRuleSet;

    constructor() {
        this.rules = {
            text: this.text.bind(this),
            tag: this.tag.bind(this),
            g: this.g.bind(this),
            strong: this.strong.bind(this),
            em: this.em.bind(this),
            s: this.s.bind(this),
            sup: this.sup.bind(this),
            samp: this.samp.bind(this),
            x: this.x.bind(this),
            code: this.code.bind(this),
            link_text_part: this.linkTextPart.bind(this),
            link_attributes_part: this.linkAttributesPart.bind(this),
            link_attributes_title: this.linkAttributesTitle.bind(this),
            link_attributes_href: this.linkAttributesHref.bind(this),
            link_reflink: this.linkRefLink.bind(this),
            link_autolink: this.linkAutolink.bind(this),
            image_text_part: this.imageTextPart.bind(this),
            image_attributes_part: this.imageAttributesPart.bind(this),
            image_attributes_src: this.imageAttributesSrc.bind(this),
            image_attributes_title: this.imageAttributesTitle.bind(this),
            image_attributes_size: this.imageAttributesSize.bind(this),
        };
    }

    render(tokens: XLFToken[]): string {
        let rendered = '';

        for (const token of tokens) {
            const handler = this.rules[token.type];
            if (handler) {
                rendered += handler(token);
            }
        }

        return rendered;
    }

    text(token: XLFToken): string {
        assert(isXLFTextToken(token));
        token as XLFTextToken;

        return token.data;
    }

    tag(token: XLFToken): string {
        const handler = this.rules[token.data];
        if (!handler) {
            return '';
        }

        return handler(token);
    }

    x(token: XLFToken): string {
        assert(isXLFTagToken(token));
        token as XLFTagToken;

        const syntax = token.syntax;
        if (!syntax?.length) {
            throw new Error("can't render g tag without syntax");
        }

        const handler = this.rules[syntax];
        if (!handler) {
            throw new Error(`syntax ${syntax} not implemented`);
        }

        return handler(token);
    }

    g(token: XLFToken): string {
        assert(isXLFTagToken(token));
        token as XLFTagToken;

        const syntax = token.syntax;
        if (!syntax?.length) {
            throw new Error("can't render g tag without syntax");
        }

        const handler = this.rules[syntax];
        if (!handler) {
            throw new Error(`syntax ${syntax} not implemented`);
        }

        return handler(token);
    }

    strong(token: XLFToken): string {
        assert(isXLFTagToken(token));
        token as XLFTagToken;
        if (!token?.equivText) {
            throw new Error(`token: ${token} missing equiv-text`);
        }

        return token.equivText ?? '**';
    }

    em(token: XLFToken): string {
        assert(isXLFTagToken(token));
        token as XLFTagToken;
        if (!token?.equivText) {
            throw new Error(`token: ${token} missing equiv-text`);
        }

        return token.equivText ?? '*';
    }

    s(token: XLFToken): string {
        assert(isXLFTagToken(token));
        token as XLFTagToken;

        if (!token?.equivText) {
            throw new Error(`token: ${token} missing equiv-text`);
        }

        return token.equivText ?? '~~';
    }

    sup(token: XLFToken): string {
        assert(isXLFTagToken(token));
        token as XLFTagToken;

        if (!token?.equivText) {
            throw new Error(`token: ${token} missing equiv-text`);
        }

        return token.equivText ?? '^';
    }

    samp(token: XLFToken): string {
        assert(isXLFTagToken(token));
        token as XLFTagToken;

        if (!token?.equivText) {
            throw new Error(`token: ${token} missing equiv-text`);
        }

        return token.equivText ?? '##';
    }

    linkTextPart(token: XLFToken): string {
        assert(isXLFTagToken(token));
        token as XLFTagToken;

        const {equivText, nodeType} = token;

        if (equivText?.length !== 2) {
            throw new Error(`token: ${token} has invalid equiv-text`);
        }

        const [open, close] = equivText.split('');

        return nodeType === 'open' ? open : close;
    }

    linkAttributesPart(token: XLFToken): string {
        assert(isXLFTagToken(token));
        token as XLFTagToken;

        const {equivText, nodeType} = token;

        if (equivText?.length !== 2) {
            throw new Error(`token: ${token} has invalid equiv-text`);
        }

        const [open, close] = equivText.split('');

        return nodeType === 'open' ? open : close;
    }

    linkAttributesHref(token: XLFToken): string {
        assert(isXLFTagToken(token));
        token as XLFTagToken;

        assert(
            token.equivText?.length,
            `x supposed to wrap original link href markup inside equiv-text`,
        );

        return token.equivText;
    }

    linkAttributesTitle(token: XLFToken): string {
        assert(isXLFTagToken(token));
        token as XLFTagToken;

        const {equivText, nodeType} = token;

        if (equivText?.length !== 2) {
            throw new Error(`token: ${token} has invalid equiv-text`);
        }

        const [open, close] = equivText.split('');

        return nodeType === 'open' ? ' ' + open : close;
    }

    linkRefLink(token: XLFToken): string {
        assert(isXLFTagToken(token));
        token as XLFTagToken;

        assert(
            token.equivText?.length,
            `x supposed to wrap original ref link markup inside equiv-text`,
        );

        return token.equivText;
    }

    linkAutolink(token: XLFToken): string {
        assert(isXLFTagToken(token));
        token as XLFTagToken;

        assert(
            token.equivText?.length,
            `x supposed to wrap original ref link markup inside equiv-text`,
        );

        return token.equivText;
    }

    imageTextPart(token: XLFToken): string {
        assert(isXLFTagToken(token));
        token as XLFTagToken;

        const {equivText, nodeType} = token;
        if (equivText?.length !== 3) {
            throw new Error(`token: ${token} has invalid equiv-text`);
        }

        const [open, close] = equivText.split(/(\]$)/gmu);
        return nodeType === 'open' ? open : close;
    }

    imageAttributesPart(token: XLFToken): string {
        assert(isXLFTagToken(token));
        token as XLFTagToken;

        const {equivText, nodeType} = token;
        if (equivText?.length !== 2) {
            throw new Error(`token: ${token} has invalid equiv-text`);
        }

        const [open, close] = equivText.split('');

        return nodeType === 'open' ? open : close;
    }

    imageAttributesSrc(token: XLFToken): string {
        assert(isXLFTagToken(token));
        token as XLFTagToken;

        if (!token?.equivText?.length) {
            throw new Error(`x supposed to wrap image src inside equiv-text`);
        }

        return token.equivText;
    }

    imageAttributesTitle(token: XLFToken): string {
        assert(isXLFTagToken(token));
        token as XLFTagToken;

        const {equivText, nodeType} = token;
        if (equivText?.length !== 2) {
            throw new Error(`token: ${token} has invalid equiv-text`);
        }

        const [open, close] = equivText.split('');

        return nodeType === 'open' ? ' ' + open : close;
    }

    imageAttributesSize(token: XLFToken): string {
        assert(isXLFTagToken(token));
        token as XLFTagToken;

        const {equivText} = token;
        if (!equivText?.length) {
            throw new Error(`token: ${token} has invalid equiv-text`);
        }

        return ' ' + equivText;
    }

    code(token: XLFToken): string {
        assert(isXLFTagToken(token));
        token as XLFTagToken;

        assert(token.equivText?.length, 'x supposed to wrap original markup inside equiv-text');

        return token.equivText;
    }
}

export {XLFMDRenderer};
export default {XLFMDRenderer};
