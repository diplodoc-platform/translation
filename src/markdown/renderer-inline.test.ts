import {RenderParameters, render} from './renderer';

describe('inline: renders translated markdown', () => {
    it('inline: renders sentences with plain text', () => {
        const parameters: RenderParameters = {
            skeleton: '%%%1%%% %%%2%%%',
            translations: new Map<string, string>([
                ['1', 'Sentence number one.'],
                ['2', 'Sentence number two.'],
            ]),
        };

        const rendered = render(parameters);
        expect(rendered).toMatchSnapshot();
    });

    it('inline: renders sentences with strong syntax', () => {
        const parameters: RenderParameters = {
            skeleton: '%%%1%%% %%%2%%%',
            translations: new Map<string, string>([
                ['1', 'Sentence number **one**.'],
                ['2', 'Sentence number **two**.'],
            ]),
        };

        const rendered = render(parameters);
        expect(rendered).toMatchSnapshot();
    });

    it('inline: renders sentences with em syntax', () => {
        const parameters: RenderParameters = {
            skeleton: '%%%1%%% %%%2%%%',
            translations: new Map<string, string>([
                ['1', 'Sentence number *one*.'],
                ['2', 'Sentence number *two*.'],
            ]),
        };

        const rendered = render(parameters);
        expect(rendered).toMatchSnapshot();
    });

    it('inline: renders sentences with s syntax', () => {
        const parameters: RenderParameters = {
            skeleton: '%%%1%%% %%%2%%%',
            translations: new Map<string, string>([
                ['1', 'Sentence number ~~one~~.'],
                ['2', 'Sentence number ~~two~~.'],
            ]),
        };

        const rendered = render(parameters);
        expect(rendered).toMatchSnapshot();
    });

    it('inline: renders sentences with sup syntax', () => {
        const parameters: RenderParameters = {
            skeleton: '%%%1%%% %%%2%%%',
            translations: new Map<string, string>([
                ['1', 'Sentence number^one^.'],
                ['2', 'Sentence number^two^.'],
            ]),
        };

        const rendered = render(parameters);
        expect(rendered).toMatchSnapshot();
    });

    it('inline: renders sentences with samp syntax', () => {
        const parameters: RenderParameters = {
            skeleton: '%%%1%%% %%%2%%%',
            translations: new Map<string, string>([
                ['1', 'Sentence number ##one##.'],
                ['2', 'Sentence number ##two##.'],
            ]),
        };

        const rendered = render(parameters);
        expect(rendered).toMatchSnapshot();
    });

    it('inline: renders sentences with code syntax', () => {
        const parameters: RenderParameters = {
            skeleton: '%%%1%%% %%%2%%%',
            translations: new Map<string, string>([
                ['1', 'Sentence number `one`.'],
                ['2', 'Sentence number `two`.'],
            ]),
        };

        const rendered = render(parameters);
        expect(rendered).toMatchSnapshot();
    });

    it('inline: renders sentences with links syntax', () => {
        const parameters: RenderParameters = {
            skeleton: '%%%1%%% %%%2%%%',
            translations: new Map<string, string>([
                ['1', 'Sentence number [one](one.md "one").'],
                ['2', 'Sentence number [two](two.md "two").'],
            ]),
        };

        const rendered = render(parameters);
        expect(rendered).toMatchSnapshot();
    });

    it('inline: renders sentences with ref links syntax', () => {
        const parameters: RenderParameters = {
            skeleton: '%%%1%%% %%%2%%%',
            translations: new Map<string, string>([
                ['1', 'Sentence number [{#T}](one.md "one").'],
                ['2', 'Sentence number [{#T}](two.md).'],
            ]),
        };

        const rendered = render(parameters);
        expect(rendered).toMatchSnapshot();
    });

    it('inline: renders sentences with autolink syntax', () => {
        const parameters: RenderParameters = {
            skeleton: '%%%1%%% %%%2%%%',
            translations: new Map<string, string>([
                ['1', 'Sentence number one <https://www.google.com>.'],
                ['2', 'Sentence number two <https://www.youtube.com>.'],
            ]),
        };

        const rendered = render(parameters);
        expect(rendered).toMatchSnapshot();
    });

    it('inline: renders sentences with variale href link syntax', () => {
        const parameters: RenderParameters = {
            skeleton: '%%%1%%% %%%2%%%',
            translations: new Map<string, string>([
                ['1', 'Sentence number [one]({{one}}).'],
                ['2', 'Sentence number [two]({{ two }}).'],
            ]),
        };

        const rendered = render(parameters);
        expect(rendered).toMatchSnapshot();
    });

    it('inline: renders sentences with variale image syntax', () => {
        const parameters: RenderParameters = {
            skeleton: '%%%1%%% %%%2%%% %%%3%%% %%%4%%% %%%5%%% %%%6%%%',
            translations: new Map<string, string>([
                ['1', 'Sentence ![image](_images/image.png "текст_подсказки" =100x100).'],
                ['2', 'Sentence ![image](_images/image.png "текст_подсказки" =x100).'],
                ['3', 'Sentence ![image](_images/image.png "текст_подсказки" =100x).'],
                ['4', 'Sentence ![image](_images/image.png =100x100).'],
                ['5', 'Sentence ![image](_images/image.png).'],
                ['6', 'Sentence ![]().'],
            ]),
        };

        const rendered = render(parameters);
        expect(rendered).toMatchSnapshot();
    });

    it('inline: renders sentences with video syntax', () => {
        const parameters: RenderParameters = {
            skeleton: '%%%1%%% %%%2%%%',
            translations: new Map<string, string>([
                ['1', 'Sentence goes here.'],
                ['2', 'Another sentence @[youtube](https://youtu.be/rJz4OaURJ6U)'],
            ]),
        };

        const rendered = render(parameters);
        expect(rendered).toMatchSnapshot();
    });

    it('inline: renders sentences with anchor heading syntax', () => {
        const parameters: RenderParameters = {
            skeleton: '%%%1%%% %%%2%%%',
            translations: new Map<string, string>([
                ['1', '# Heading with multiple anchors {#anchor1} {#anchor2}'],
                ['2', '## Heading without anchor'],
            ]),
        };

        const rendered = render(parameters);
        expect(rendered).toMatchSnapshot();
    });

    it('inline: renders sentences with file syntax', () => {
        const parameters: RenderParameters = {
            skeleton: '%%%1%%% %%%2%%%',
            translations: new Map<string, string>([
                [
                    '1',
                    'Sentence with {% file src="path/to/file" name="readme.md" referrerpolicy="no-referrer" rel="noopener" target="_blank" type="text/plain" %}.',
                ],
                ['2', 'Another sentence.'],
            ]),
        };

        const rendered = render(parameters);
        expect(rendered).toMatchSnapshot();
    });

    it('inline: renders sentences with liquid conditions', () => {
        const parameters: RenderParameters = {
            skeleton: '%%%1%%% %%%2%%%',
            translations: new Map<string, string>([
                ['1', 'Sentence with {% if var == "val" %} val {% else %} other val {% endif %}.'],
                ['2', '{% var == "val" %} A {% else %} B {% endif %} Point.'],
            ]),
        };

        const rendered = render(parameters);
        expect(rendered).toMatchSnapshot();
    });
});
