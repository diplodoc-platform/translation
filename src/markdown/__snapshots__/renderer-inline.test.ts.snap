// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`inline: renders translated markdown inline: renders sentences with code syntax 1`] = `"Sentence number \`one\`. Sentence number \`two\`."`;

exports[`inline: renders translated markdown inline: renders sentences with em syntax 1`] = `"Sentence number *one*. Sentence number *two*."`;

exports[`inline: renders translated markdown inline: renders sentences with plain text 1`] = `"Sentence number one. Sentence number two."`;

exports[`inline: renders translated markdown inline: renders sentences with s syntax 1`] = `"Sentence number ~~one~~. Sentence number ~~two~~."`;

exports[`inline: renders translated markdown inline: renders sentences with samp syntax 1`] = `"Sentence number ##one##. Sentence number ##two##."`;

exports[`inline: renders translated markdown inline: renders sentences with strong syntax 1`] = `"Sentence number **one**. Sentence number **two**."`;

exports[`inline: renders translated markdown inline: renders sentences with sup syntax 1`] = `"Sentence number^one^. Sentence number^two^."`;
